src/main/java/reposense/report/ReportGenerator.java

```diff
@@ -118,9 +126,24 @@
      * Copies the template file to the specified {@code outputPath} for the repo report to be generated.
      * @throws IOException if template resource is not found.
      */
-    private static void prepareTemplateFile(String outputPath) throws IOException {
+    private static void prepareTemplateFile(ReportConfiguration config, String outputPath) throws IOException {
         InputStream is = RepoSense.class.getResourceAsStream(TEMPLATE_FILE);
         FileUtil.copyTemplate(is, outputPath);
+        setReportConfiguration(config, outputPath);
+    }
+
+    private static void setReportConfiguration(ReportConfiguration config, String outputPath) throws IOException {
+        setLandingPageTitle(outputPath, config.getTitle());
+    }
+
+    /**
+     * Set title of template file located at {@code filePath} to {@code pageTitle}
+     */
+    private static void setLandingPageTitle(String filePath, String pageTitle) throws IOException {
```

Comment:
Can we set the page title using Vue's functionality? Like we pass the title in the JSON file so the Vue can read and update it?

Reply:
I agree that in the case of pageTitle, this way may be better. But, in the case that we need more customized settings, we still need to pass them to frontend using JSON.

Reply:
Or we can seek to generate the Vue config file using the given parameters, including the title, and then build the Vue project. [Link](https://forum.vuejs.org/t/how-to-pass-command-line-parameters-to-vue/38701)

Reply:
Yes, I think it is possible to let Vue read in page title and and customize it easily even after report generation. I don't have a preference on usage of a custom json file or vue config right now though, but I will dig a little deeper and update this PR accordingly.

Reply:
![ezgif com-optimize](https://user-images.githubusercontent.com/35420846/98391232-58608780-2091-11eb-9ec3-3662522ef60c.gif)
Here is an example of how dynamically loading a title from a JSON file in Vue is like. Basically, there can be a noticeable delay in the loading of page titles. I dont think there is a suitable Vue lifecycle event that allows us to inject a page title without this delay.

Reply:
Also regarding vue config, and if I did my research right, vue config is used together with Vue CLI, which is mainly used for development purposes, and I’m not really comfortable with adding such a requirement on the users for the running of RepoSense.

Reply:
I’ve also tried looking up on other solutions, but they are also mainly designed around web dev too.
Maybe someone has a better suggestion on how a page title can be changed via frontend means?

Reply:
Can't the config be used during building of vue project?

Reply:
Correct me if I’m wrong. But i dont think running reposense right now, involves an extra build step for the building of the vue project, and most of the required templates/js/css files have already been pre-generated and aggregated into the JAR for usage?

Reply:
Also, when we talk about building a vue project. do you mean letting the users build the project themselves? Or do you mean abstracting the build process within the executable?

Reply:
Technically, we have a self-defined building step for frontend called `spuild` (you can check in npm run). It is meant to be deprecated, but is still used now. In the near future, we are going to turn our frontend into a formal vue project as describe in #795.

Reply:
I will also check the logic of Jar file.

Reply:
Yes, you are correct. When running the Jar file, we don't need to build the report again. Seems that there are only two options remaining. One is to edit the title directly. The second is the dynamic loading. 

I am not sure if Prof is OK if we can view a noticeable delay in the loading of page titles. We may first show a `RepoSense Report` title rather than `Example Sample` at the first step of loading. So the following change of title won't cause much misunderstanding.

@damithc  The situation is like this, we want to pass the title defined in a user-defined config to frontend. And there are two ways.
1. We modify the relative part in the HTML code directly. But the extensibility is lower and we cannot extend it to further attributes.

2. We dynamically load the attribute from a JSON. But we can view a noticeable delay in the loading of page titles. The disadvantage maybe somehow mitigated by using a `RepoSense Report` as default title.

Reply:
@damithc Prof, your input is needed here.

Reply:
Sorry, didn't notice it before. 
For 1, what are other attributes we could extend to? If we can't think of any, we can go with 1.
I'm OK with 2 too.

Reply:
Currently, I dont have any plans to extend this functionality. But using html parsers can be a viable solution if we want to extend it to other attributes. But might be too complex given how simple our use case is right now?

Reply:
I see. I'm OK with both options. You guys can choose.

Reply:
The related functionality is mentioned in #467. Then, let's go with option 1.